src/lib.rs

```diff
@@ -53,7 +57,7 @@ mod io {
         }
     }
 
-    pub type Result<T> = core::result::Result<T, Error>;
+    pub(crate) type Result<T> = core::result::Result<T, Error>;
```

Comment:
is this change necessary?

Reply:
Tried to revert this and the answer is yes - linter complains about it being an unreachable pub item

<img width="944" alt="image" src="https://github.com/user-attachments/assets/561b4153-ff36-44d3-b87d-c6560c30c0c0">